tests/integration-test/src/tests/utils.rs

```diff
@@ -493,84 +568,98 @@ pub fn add_fentry_or_fexit(
     }
 
     if fentry {
-        args.extend(["fentry", "-f", "do_unlinkat"]);
+        args.extend(["fentry", "-f", fn_name]);
     } else {
-        args.extend(["fexit", "-f", "do_unlinkat"]);
+        args.extend(["fexit", "-f", fn_name]);
     }
 
-    let output = Command::cargo_bin("bpfman")?.args(args).ok();
-    let stdout = String::from_utf8(output.unwrap().stdout).unwrap();
-    let prog_id = bpfman_output_parse_id(&stdout);
-    assert!(!prog_id.is_empty());
-    if fentry {
-        debug!(
-            "Successfully added fentry program: {:?} from: {:?}",
-            prog_id, load_type
-        );
-    } else {
-        debug!(
-            "Successfully added fexit program: {:?} from: {:?}",
-            prog_id, load_type
-        );
+    match call_bpfman(args) {
+        Ok(stdout) => {
+            let prog_id = bpfman_output_parse_id(&stdout);
+            assert!(!prog_id.is_empty());
+            if fentry {
+                debug!(
+                    "Successfully added fentry program: {:?} from: {:?}",
+                    prog_id, load_type
+                );
+            } else {
+                debug!(
+                    "Successfully added fexit program: {:?} from: {:?}",
+                    prog_id, load_type
+                );
+            }
+            Ok(prog_id)
+        }
+        Err(e) => Err(e),
     }
-    Ok(prog_id)
 }
 
 /// Delete a bpfman program using bpfman
-pub fn bpfman_del_program(prog_id: &str) {
-    Command::cargo_bin("bpfman")
-        .unwrap()
-        .args(["unload", prog_id.trim()])
-        .assert()
-        .success()
-        .stdout(is_empty());
-
-    debug!("Successfully deleted program: \"{}\"", prog_id.trim());
+pub fn bpfman_del_program(prog_id: &str) -> Result<()> {
```

Comment:
We could hid the assert!(res.is_ok()) in here right? Then we avoid doing 

```
    let result = bpfman_del_program(&(shared_owner_id.unwrap()));
    assert!(result.is_ok());
``` 

everwhere

Reply:
Then I couldn't use `bpfman_del_program()` for the negative testing.